rasa_sdk/knowledge_base/actions.py

```diff
@@ -0,0 +1,372 @@
+# -*- coding: utf-8 -*-
+from rasa_sdk import Action
+from rasa_sdk.events import SlotSet
+from rasa_sdk.knowledge_base.storage import (
+    SCHEMA_KEYS_ATTRIBUTES,
+    SCHEMA_KEYS_REPRESENTATION,
+    SCHEMA_KEYS_KEY,
+)
+
+
+SLOT_MENTION = "mention"
+SLOT_OBJECT_TYPE = "object_type"
+SLOT_ATTRIBUTE = "attribute"
+SLOT_LISTED_OBJECTS = "knowledge_base_listed_objects"
+SLOT_LAST_OBJECT = "knowledge_base_last_object"
+SLOT_LAST_OBJECT_TYPE = "knowledge_base_last_object_type"
+
+
+class ActionKnowledgeBase(Action):
+    """
+    Abstract knowledge base action that can be inherited to create custom actions
+    that are able to interact with a knowledge base.
+    """
+
+    def __init__(self, knowledge_base):
+        self.knowledge_base = knowledge_base
```

Comment:
if I remember correctly, action is init every time, shouldn't this knowledge_base be a part of the tracker?

Reply:
Not sure about when an action is init. But adding knowledge_base to the tracker would complicate things quite a bit. Maybe @tmbo or @wochinge knows.

Reply:
I remember when we did `FormAction` one of the problems was that it should be stateless

Reply:
I discussed it with @tmbo. Should not be a problem. The action is initialized just once.

Reply:
wait, so custom action can contain instance variables?

Reply:
We discussed it with tmbo, now I see why it's fine and instance variables are not